homeassistant/components/deconz/binary_sensor.py

```diff
@@ -120,30 +135,32 @@ class DeconzBinarySensor(DeconzDevice, BinarySensorEntity):
     """Representation of a deCONZ binary sensor."""
 
     TYPE = DOMAIN
+    _device: PydeconzBinarySensor
 
-    def __init__(self, device, gateway):
+    def __init__(self, device: PydeconzBinarySensor, gateway: DeconzGateway) -> None:
         """Initialize deCONZ binary sensor."""
         super().__init__(device, gateway)
 
         if entity_description := ENTITY_DESCRIPTIONS.get(type(device)):
             self.entity_description = entity_description
 
     @callback
-    def async_update_callback(self):
+    def async_update_callback(self) -> None:
         """Update the sensor's state."""
         keys = {"on", "reachable", "state"}
         if self._device.changed_keys.intersection(keys):
             super().async_update_callback()
 
     @property
-    def is_on(self):
+    def is_on(self) -> bool:
         """Return true if sensor is on."""
+        assert isinstance(self._device.state, bool)
         return self._device.state
 
     @property
-    def extra_state_attributes(self):
+    def extra_state_attributes(self) -> dict:
         """Return the state attributes of the sensor."""
-        attr = {}
+        attr: dict[str, bool | float | int | str | list | None] = {}
```

Comment:
I couldn't find anything that returned str, is it needed? (Could be that I missed something!)

Reply:
Thanks!